### DISCUSSION TITLE
Starting Multiple Microcontrollers in AP mode and automating SSID/PW replies.

### DISCUSSION THREAD

**Discussion by GrahamWillsher** [khoih-prog/ESP_WiFiManager_Lite#5] [Ideas]
Hi,

Please excuse me contacting you and I hope that you don't mind the newbie questions.

I have downloaded and experimented with your code and it solves a lot of the problems that I had.

- Is there anyway that, during initial AP mode, the code can check to see if another board is not already in the same initialisation AP mode? The reason that I ask is that my project is going to have a number of Microprocessor boards all starting up at 'potentially' the same time which will lead to mulitple Access Points with the same credentials. I was thinking of something simple like collision detection e.g. see if an access point of name 'xxx' is running, if there is wait for a random amount of time then check again, if there is no access point start in AP mode.
- Also is there a simple way to send the SSID and Password of the required network back to the AP from a computer program rather than having to enter the credentials manually? The reason I ask is I will have mutliple microprocessors starting up and there is potential to get the credentials wrong if they are manually entered.

Thanks for your help in this matter, and I am sorry if the answers are already in your code and I have missed them, if so, please could you point me in the right direction.

Regards,

Graham.

**Comment by khoih-prog** [comment]
> Is there anyway that, during initial AP mode, the code can check to see if another board is not already in the same initialisation AP mode? The reason that I ask is that my project is going to have a number of Microprocessor boards all starting up at 'potentially' the same time which will lead to mulitple Access Points with the same credentials. I was thinking of something simple like collision detection e.g. see if an access point of name 'xxx' is running, if there is wait for a random amount of time then check again, if there is no access point start in AP mode.

Please check the code [](https://github.com/khoih-prog/ESP_WiFiManager_Lite/blob/main/examples/ESP_WiFi/ESP_WiFi.ino#L79-L80) and see that you can customized the AP credentials for each board. Just use some simple function to change the SSIS / PWD according to some string with time, date, etc.

```
String AP_SSID = "your_customized_ssid";
String AP_PWD  = "your_customized_pwd";
```

The examples are just examples to show you how to use basic features, and can't never be perfectly applied in all use cases, especially in your complex use-case.

> Also is there a simple way to send the SSID and Password of the required network back to the AP from a computer program rather than having to enter the credentials manually? The reason I ask is I will have mutiple microprocessors starting up and there is potential to get the credentials wrong if they are manually entered.

Anything is possible, certainly if you write the program to do so. Check [**So, how it works?**](https://github.com/khoih-prog/ESP_WiFiManager_Lite#so-how-it-works), and post `your request for help` on some forums, such as Arduino, ESP32/ESP8266. Normally, the WebBrowers can remember and suggest the input using your recent data.

Also you have to enable `SCAN_WIFI_NETWORKS`, as in [**defines.h#L92**](https://github.com/khoih-prog/ESP_WiFiManager_Lite/blob/main/examples/ESP_WiFi/defines.h#L92) and have the SSID autodisplay, such as in [**With SCAN_WIFI_NETWORKS**](https://github.com/khoih-prog/ESP_WiFiManager_Lite#2-with-scan_wifi_networks)

I'm sorry I certainly have no time to help you out. Hopefully someone reading this can help guide you to write the code.

Remember you need to use, test and familiarize yourself with the operation of the library, before taking on the complex projects with many boards.

Good Luck,

**Reply by rmallah** [reply]
Hi @GrahamWillsher 

below code will help you to set the AP name based on MAC
this will set a unique name for each AP and prevent collision.

```
  String macaddress = WiFi.macAddress();
  macaddress.replace(":", "");
  
  Serial.println( macaddress);
    
  String AP_SSID = "YourAP-Prexfix-" + macaddress ; 
  String AP_PWD  = "password";

```

**Reply by GrahamWillsher** [reply]
Hi rmallah,

Thanks for the reply,  another member aso suggested the same solution which worked as required.

I also managed to work out what fields were required so that I could automatically have code on another machine look for the necessary 'AP' and automatically send the login information to it so that I did not have to manually log into the 'AP' via a browser.

All in all it worked perfectly.

Thanks again for your reply.

Regards,
Graham.

**Reply by rmallah** [reply]
Thanks for the kind email. I am glad that things worked from community help.
Thanks to the project maintainer for creating such a gem with multiple reusable
libraries and examples.
